### ISSUE TITLE
Publish source mapping files to CDN

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#9076]
E.g. 

> //# sourceMappingURL=bokeh.js.map

This (AFAIK) will suppress the annoying 404s users see and that also show up in the CDN logs. 

cc @mattpap

**Comment by bryevdv** [comment]
I can imagine doing this postproc with `sed` or `perl` but I think it would be cleaner to have it be an option passed somehow to `node make` that is conditionally set on full release builds.

**Comment by mattpap** [comment]
`*.map` file references shouldn't be an issue unless someone opens up devtools, in which case it would be actually good to have them available for ad-hoc debugging. Based on a cursory check, I don't see them being available on the CDN. Perhaps we should upload them instead?

**Comment by bryevdv** [comment]
>  Based on a cursory check, I don't see them being available on the CDN. Perhaps we should upload them instead?

Right, they are not on the CDN, we could certainly start to upload them as another option if that is useful. I don't have a clear/strong opinion here.

**Comment by mattpap** [comment]
It would be useful, because I already needed to debug bokeh on-line several times, where minified resources were used, and only options were either to reproduce things locally (not always possible) or to try debugging minified (or half pretty printed) code, which is no fun.

**Comment by bryevdv** [comment]
Have had second thoughts about this. It is a huge bookkeeping PITA, adds risk to automated release machinery, and will explosively clutter the already very cluttered "everything in one prefix with versions in the filename" S3 layout.

**Comment by bryevdv** [comment]
> *.map file references shouldn't be an issue unless someone opens up devtools,

This is clearly not the case we have hundreds of thousands of 404s registered on the CDN for each of these files. As well as user confusion:

https://discourse.bokeh.org/t/why-flask-need-bokeh-min-js-map-files-for-bokeh-1-3-4/4169/2

Starting with 2.0 I would really just like to strip these from the published artifacts.

**Comment by mattpap** [comment]
> we have hundreds of thousands of 404s registered on the CDN for each of these files.

If we don't upload those files, then 404s are perfectly understandable. There are a few options to resolve this:

1. Drop minified sources map altogether.
2. Upload source maps to CDN.
3. Upload stripped js files to CDN.

So, (1) is wrong and will hinder development, (2) would be a decent options, but then including version number in file names will be a problem (it's not a good idea in general, but I guess it's a limitation of using S3) and (3) is not ideal but workable. However, if you find your self in a situation that you have to debug bokehjs, and you can't alter resources, then you will have to rely on devtools' pretty printing capabilities and debugging like this is pretty painful.

> https://discourse.bokeh.org/t/why-flask-need-bokeh-min-js-map-files-for-bokeh-1-3-4/4169/2

This looks like a configuration issue, and doesn't seem to be related to CDN.

**Comment by bryevdv** [comment]
> This looks like a configuration issue, and doesn't seem to be related to CDN.

My point about the CD nNumbers is that *lots and lots and lots* of people are evidently having their browsers ask for these files, all the time. I would be really surprised if that many people have dev tools turned on. I think 3) is OK, the situation you describe just does not come up that often and all these requests are both a drain on the CDN as well as extra overhead on the user end.

**Comment by bryevdv** [comment]
Just noting that I have had yet another support interaction today where someone thought these 404s were the source of their problem. And it was not evident at first that these were what they were referring to, so there was a whole back and forth just to get to the point of "those error messages are annoying but harmless". Which is also not really a re-assuring thing for users to hear. 

This has been an issue and question repeatedly over the years, whereas I have never once needed map files on a full release where I couldn't just run locally. I think we have to strip these comments out of the full release builds.

**Comment by bryevdv** [comment]
We can certainly discuss, but I don't want to forget to discuss, so I am moving this into the milestsone.

**Comment by mattpap** [comment]
I'm actually fine with stripping those. We just have to make sure that internally we only use development resources, which at this point is at least mostly true.
